src/components/Typography/Typography.js

```diff
@@ -0,0 +1,98 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+
+
+export const Typography = (props) => {
+  const {
+    className,
+    children,
+    variant,
+    color,
+    ...otherProps
+  } = props
+
+
+  const colorFont = color
+
+
+if (variant === 'h1') {
```

Comment:
Sporo rzeczy się tutaj powiela.
Może da się to zrobić w taki sposób: https://devmentor.pl/b/dynamiczne-renderowanie-komponentow-w-react